zebra-state/src/service/chain_tip.rs

```diff
@@ -387,6 +391,38 @@ impl ChainTip for LatestChainTip {
         self.with_chain_tip_block(|block| block.transaction_hashes.clone())
             .unwrap_or_else(|| Arc::new([]))
     }
+
+    /// Returns when the state tip changes.
+    ///
+    /// Marks the state tip as seen.
```

Comment:
```suggestion
    /// Marks the state tip as seen when called (but not when BestTipChanged is polled)
```

Reply:
You're right, I think I needed the old design when I required a `'static` lifetime in the `BestTipChanged` struct. But now it has a lifetime parameter, it's ok for it to depend on the lifetime of the receiver.

This is in e72d0ebac4, it simplifies the production and test code a lot.